### ISSUE TITLE
performance degradation (time) of insert / upsert in batch operation

### ISSUE THREAD

**Issue by effesessa** [arangodb/arangodb#12311]
## My Environment

* __ArangoDB Version__:        3.6.5
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:   Kubernetes
* __Configuration__:               
```yaml
apiVersion: "database.arangodb.com/v1alpha"
kind: "ArangoDeployment"
metadata:
  name: "cluster"
spec:
  externalAccess:
    type: NodePort
  auth:
    jwtSecretName: None
  tls:
    caSecretName: None
  mode: Cluster
  image: arangodb/arangodb:3.6.5
  dbservers:
    count: 3
    args: ["--rocksdb.write-buffer-size=1024MB", "--rocksdb.max-write-buffer-number=6"]
    overrideDetectedTotalMemory: true
  coordinators:
    count: 3
    overrideDetectedTotalMemory: true
```
* __Infrastructure__:               Azure
* __Total RAM in your machine__:        16GB for each node (3 worker nodes)
* __Disks in use__:        SSD 8GB PVC for each pod
* __Used Package__:             Docker - official Docker library | other 

## Component, Query & Data

__Affected feature__:
AQL INSERT/UPSERT IN BATCH OPERATION (STRESS)

__Replication Factor & Number of Shards (Cluster only)__:
3 & 3

## Steps to reproduce
```java
// bomb-arangodb jar
...
queries.forEach(query -> {
    arangoDatabase.query(query, String.class);
});
arangoDB.shutdown();
...
```

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: bomb-arangodb-write-deployment
  labels:
    app: bomb-arangodb-write-deployment
spec:
  replicas: 20
  selector:
    matchLabels:
      app: bomb-arangodb-write
  template:
    metadata:
      labels:
        app: bomb-arangodb-write
    spec:
      containers:
        - name: bomb-arangodb-write-container
          image: <<ACR_NAME>> .azurecr.io/bomb-arangodb:0.0.3
          imagePullPolicy: Always
          ports:
            - containerPort: 8080
          command: ["/bin/bash"]
          args: ["-c", "java -cp bomb-arangodb.jar com.effe.WriteApp 25000 cluster-ea.arango.svc.cluster.local"]
```

__Problem__:
I start 20 pods (as posted in the deployment above, then 25000 writes per head = tot: 500000) that write ["insert" or "upsert"] (both tested) and as time passes I see a degradation of the times of every single write (moreover, sometimes some dbserver or coordinator node falls). I don't really know how to fix this. I also increased the size of the write buffer assuming a possible problem but nothing always the same story.

times go from 20 milliseconds to 300 400 dopo circa 100k-200k writes - up to 1000 milliseconds when it reaches 400k writes and so on

Please, point me in the right way. thank you

__Expected result__:
1) No time degradation
2) No fall of the nodes

**Comment by graetzer** [comment]
So assuming the compaction kicks in the performance may decrease until the level0 is free again. Did you provide the code of your stress test? Are you using indexes? These may increase insert times when doing updates

Have you ever tried increasing the level0 values: 
`--rocksdb.level0-slowdown-trigger 30 --rocksdb.level0-stop-trigger 50`

Perhaps increasing the target size would also help: `--rocksdb.target-file-size-base 256MiB`

https://www.arangodb.com/docs/stable/data-modeling-operational-factors.html

**Comment by graetzer** [comment]
Perhaps you could provide us at witht the code of your stress test

**Comment by Simran-B** [comment]
@effesessa Did you have a chance to try out graetzer's suggestions?

**Comment by Simran-B** [comment]
Looks like the suggested configuration changed solved #12190, therefore I'm also closing this issue.
